modules/cloud-auth/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+if(ENABLE_CLOUD_AUTH AND NOT ENABLE_CPP)
+  message(FATAL_ERROR "C++ support is mandatory when the Cloud Auth module is enabled.")
+endif()
+
+module_switch(ENABLE_CLOUD_AUTH "Enable Cloud Auth" ENABLE_CPP)
```

Comment:
I copied this from other modules, and it seems to work, yes.

Reply:
the idea behind module_switch is that it will define e.g ENABLE_CLOUD_AUTH if it is not already defined that way you can control options like this from the cmake command line as well e.g. -DENABLE_CLOUD_AUTH=OFF